engine/schema/src/main/java/com/cloud/storage/dao/VolumeDaoImpl.java

```diff
@@ -365,6 +375,12 @@ public VolumeDaoImpl() {
         AllFieldsSearch.and("name", AllFieldsSearch.entity().getName(), Op.EQ);
         AllFieldsSearch.done();
 
+        RootDiskStateSearch = createSearchBuilder();
```

Comment:
May be you can re-use _AllFieldsSearch_, by changing Op.EQ to Op.IN here: https://github.com/apache/cloudstack/blob/67bc74de56357eeefdec9cccb6314d73dc757f5d/engine/schema/src/main/java/com/cloud/storage/dao/VolumeDaoImpl.java#L362

Reply:
@sureshanaparti I was thinking about it. but at the end I decided to add a new searchbuilder.
no idea which way is suggested.

Reply:
@weizhouapache suggested to re-use _AllFieldsSearch_ , but it is ok with new searchbuilder as well.